processor/tailsamplingprocessor/internal/decisioncache/lru_cache.go

```diff
@@ -0,0 +1,42 @@
+package decisioncache
+
+import (
+	"encoding/binary"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"sync"
+)
+
+// lruDecisionCache implements DecisionCache as a simple LRU cache
+type lruDecisionCache struct {
+	cache *lru.Cache[uint64, bool]
+	sync.RWMutex
+}
+
+var _ DecisionCache = (*lruDecisionCache)(nil)
+
+func NewLRUDecisionCache(size int, onEvict func(uint64, bool)) (DecisionCache, error) {
```

Comment:
onEvict is always an empty function. Perhaps it shouldn't be there? We can add it later if we need.

Reply:
Yes, happy to leave that empty and we can add evict capabilities later if we wish.